Unscheduled Care/2. Unscheduled Care outputs.R

```diff
@@ -333,31 +335,87 @@ EAs_loc_ts <- area_trend_usc(
 )
 
 
-# Objects for text and summary table
+#Objects for text for Figures and Summary (Emergency Admissions)
+min_year_ea <- min(emergency_adm_areas$financial_year)
+max_year_ea <- max(emergency_adm_areas$financial_year)
+
+first_fy_rate <- filter(
+  emergency_adm_areas,
+  (financial_year == min(emergency_adm_areas$financial_year)) &
+    (location == LOCALITY & area_type == "Locality")
+)$data
+
 latest_emergency_adm_loc <- emergency_adm_areas %>%
   filter(
     location == LOCALITY,
-    year == max(year)
+    year == max(year, na.rm = TRUE)
   ) %>%
-  mutate(data = format(data, big.mark = ",")) %>%
-  pull(data)
+  mutate(data2 = format(data, big.mark = ",")) #%>%
+  #pull(data)
+
+latest_emergency_adm_loc1 <- latest_emergency_adm_loc %>% pull(data2)
+latest_emergency_adm_loc2 <- latest_emergency_adm_loc %>% pull(data)
 
+percent_rate_change <- round(abs(latest_emergency_adm_loc2 - first_fy_rate) / first_fy_rate * 100, digits = 1)
```

Comment:
This crops up a lot and might benefit from being a function.

```r
good_function_name <- function(numerator, denominator) {
    round(abs(numerator - denominator) / denominator* 100, digits = 1)
}
```

```suggestion
percent_rate_change <- good_function_name(latest_emergency_adm_loc2, first_fy_rate)
```